djstripe/stripe_objects.py

```diff
@@ -1273,6 +1277,54 @@ def create_token(cls, number, exp_month, exp_year, cvc, **kwargs):
         return stripe.Token.create(card=card)
 
 
+class StripeCoupon(StripeObject):
+
+    DURATION_FOREVER = "forever"
+    DURATION_ONCE = "once"
+    DURATION_REPEATING = "repeating"
+
+    DURATIONS = [DURATION_FOREVER, DURATION_ONCE, DURATION_REPEATING]
+    DURATION_CHOICES = [(duration, duration.replace("_", " ").title()) for duration in DURATIONS]
+
+    class Meta:
+        abstract = True
+        unique_together = ("stripe_id", "livemode")
+
+    stripe_class = stripe.Coupon
+    stripe_dashboard_item_name = "coupons"
+
+    stripe_id = StripeIdField(stripe_name="id", max_length=500)
+    amount_off = StripeCurrencyField(
+        null=True, blank=True,
+        help_text="Amount that will be taken off the subtotal of any invoices for this customer."
+    )
+    currency = StripeCharField(null=True, blank=True, max_length=3, help_text="Three-letter ISO currency code")
+    duration = StripeCharField(
+        max_length=9, choices=DURATION_CHOICES,
+        help_text="Describes how long a customer who applies this coupon will get the discount."
+    )
+    duration_in_months = StripePositiveIntegerField(
+        null=True, blank=True,
+        help_text="If `duration` is `repeating`, the number of months the coupon applies."
+    )
+    max_redemptions = StripePositiveIntegerField(
+        null=True, blank=True,
+        help_text="Maximum number of times this coupon can be redeemed, in total, before it is no longer valid."
+    )
+    percent_off = StripePositiveIntegerField(
+        null=True, blank=True, validators=[MinValueValidator(1), MaxValueValidator(100)]
```

Comment:
The reason this is not a StripePercentField, by the way, is that I confirmed this is purely an integer with the possible values 1 through 100. Decimals are not possible.